payments-core/src/main/java/com/stripe/android/googlepaylauncher/GooglePayPaymentMethodLauncherViewModel.kt

```diff
@@ -1,40 +1,51 @@
 package com.stripe.android.googlepaylauncher
 
 import android.app.Application
+import android.os.Bundle
 import androidx.annotation.VisibleForTesting
+import androidx.lifecycle.AbstractSavedStateViewModelFactory
 import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
-import androidx.lifecycle.ViewModelProvider
 import androidx.lifecycle.distinctUntilChanged
+import androidx.savedstate.SavedStateRegistryOwner
 import com.google.android.gms.tasks.Task
 import com.google.android.gms.wallet.PaymentData
 import com.google.android.gms.wallet.PaymentDataRequest
 import com.google.android.gms.wallet.PaymentsClient
-import com.stripe.android.GooglePayConfig
 import com.stripe.android.GooglePayJsonFactory
 import com.stripe.android.Logger
 import com.stripe.android.PaymentConfiguration
 import com.stripe.android.exception.APIConnectionException
 import com.stripe.android.exception.InvalidRequestException
+import com.stripe.android.googlepaylauncher.injection.DaggerGooglePayPaymentMethodLauncherViewModelFactoryComponent
+import com.stripe.android.googlepaylauncher.injection.GooglePayPaymentMethodLauncherViewModelSubcomponent
 import com.stripe.android.model.PaymentMethodCreateParams
-import com.stripe.android.networking.AnalyticsRequestFactory
 import com.stripe.android.networking.ApiRequest
-import com.stripe.android.networking.StripeApiRepository
 import com.stripe.android.networking.StripeRepository
-import kotlinx.coroutines.Dispatchers
+import com.stripe.android.payments.core.injection.Injectable
+import com.stripe.android.payments.core.injection.WeakMapInjectorRegistry
 import kotlinx.coroutines.flow.first
 import org.json.JSONObject
-import kotlin.coroutines.CoroutineContext
+import javax.inject.Inject
 
-internal class GooglePayPaymentMethodLauncherViewModel(
+internal class GooglePayPaymentMethodLauncherViewModel @Inject constructor(
     private val paymentsClient: PaymentsClient,
     private val requestOptions: ApiRequest.Options,
     private val args: GooglePayPaymentMethodLauncherContract.Args,
     private val stripeRepository: StripeRepository,
     private val googlePayJsonFactory: GooglePayJsonFactory,
-    private val googlePayRepository: GooglePayRepository
+    private val googlePayRepository: GooglePayRepository,
+    private val savedStateHandle: SavedStateHandle
 ) : ViewModel() {
-    var hasLaunched: Boolean = false
+    /**
+     * [hasLaunched] indicates whether Google Pay has already been launched, and must be persisted
+     * across process death in case the Activity and ViewModel are destroyed while the user is
+     * interacting with Google Pay.
+     */
+    var hasLaunched: Boolean
```

Comment:
`internal` ?

Reply:
The class is internal, so it's internal by default. Added anyway, I guess we don't have a convention for this.